packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.test.ts

```diff
@@ -0,0 +1,170 @@
+
+const CUSTOMER_ID = 'CUSTOMER_ID'
+const BASKET_ID = 'BASKET_ID'
+
+jest.mock('../../auth/index.ts', () => {
+    return jest.fn().mockImplementation(() => ({
+        ready: jest.fn().mockResolvedValue({access_token: '123'})
+    }))
+})
+
+jest.mock('../useCustomerId.ts', () => {
+    return jest.fn().mockReturnValue(CUSTOMER_ID)
+})
+
+// TODO: should this be in test-utils? So it can be reused across test suites
+const mockMutationEndpoints = (matchingPath: string, options?: {errorResponse: number}) => {
+    const responseStatus = options?.errorResponse ? options.errorResponse : 200
+
+    nock(DEFAULT_TEST_HOST)
+        .patch((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(responseStatus, {test: 'new data'})
+        .put((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(responseStatus, {test: 'new data'})
+        .post((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(responseStatus, {test: 'new data'})
+        .delete((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(options?.errorResponse ? options.errorResponse : 204)
+}
+
+type MutationPayloads = {
+    [key in ShopperBasketMutationType]?: {body: any; parameters: any}
+}
+const mutationPayloads: MutationPayloads = {
+    updateBasket: {parameters: {basketId: BASKET_ID}, body: {}},
+    updateBillingAddressForBasket: {parameters: {basketId: BASKET_ID}, body: {}},
+    deleteBasket: {parameters: {basketId: BASKET_ID}, body: {}}
+
+    // TODO: add more payloads
+}
+
+const tests = (Object.keys(mutationPayloads) as ShopperBasketMutationType[]).map((mutationName) => {
+    const payload = mutationPayloads[mutationName]
+
+    return {
+        hook: mutationName,
+        cases: [
+            {
+                name: 'success',
+                assertions: async () => {
+                    mockMutationEndpoints('/checkout/shopper-baskets/')
+
+                    // for get basket
+                    nock(DEFAULT_TEST_HOST)
+                        .persist()
+                        .get((uri) => {
+                            return uri.includes('/checkout/shopper-baskets/')
+                        })
+                        .reply(200, {test: 'old data'})
+
+                    // for get customer basket
+                    nock(DEFAULT_TEST_HOST)
+                        .persist()
+                        .get((uri) => {
+                            return uri.includes('/customer/shopper-customers/')
+                        })
+                        .reply(200, {test: 'old data'})
+                        .get((uri) => {
+                            return uri.includes('/customer/shopper-customers/')
+                        })
+                        .reply(200, {test: 'new data'})
+
+                    const {result, waitForValueToChange} = renderHookWithProviders(() => {
```

Comment:
Kevin started using `renderHook` (instead of `render`) in his recent PR. So I'm following the trend here too.